packages/devtools_app/lib/src/config_specific/import_export/import_export.dart

```diff
@@ -126,6 +126,20 @@ abstract class ExportController {
     return '${prefix}_$timestamp$postfix.$type';
   }
 
+  /// Downloads a file with [contents] and pushes notification about success.
+  void downloadAndNotify(
```

Comment:
instead of creating a new method, can we add a parameter 'notify' to downloadFile, and then conditionally call `notificationService.push(successfulExportMessage(file));`?

Reply:
downloadFile is overriden for each platform, so created one more method to avoid duplicating this logic many times.

Reply:
You can put this in the superclass method and then annotate with mustCallSuper so that subclasses have to call `super.downloadFile`

Reply:
Then we will need to also add 'mustBeOverridden' and it will be not very clear what is already implemented in super and what is not yet.

I separated methods instead: 
- saveFile just saves file and is different for each platform
- downloadFile wraps saveFile with additional logic that is platform agnostic